### PR TITLE
Motors test mode switch and dynamically enabling/disabling sliders

### PR DESCRIPTION
**Pull request by tricopterY**
Only tested against 'allowTestMode' and Not 'feature3DSupported'
@hydra - please check to see if this PR addresses  #297 and #233
Many thanks

### CODE REVIEW

**Comment by skaman82** [comment]
@tricopterY Tested with 1.10 and 1.12 on a naze rev5. Works as expected. The 3D bug is gone now.

File: tabs/motors.html [hunk 1]
```diff
@@ -118,7 +118,7 @@
                     </div>
                     <div class="notice">
                         <p i18n="motorsNotice"></p>
-                        <label><input id="motorsEnableTestMode" type="checkbox" /><span
+                        <label><input id="motorsEnableTestMode" type="checkbox" class="toggle"/><span
```
Comment (skaman82) [inline tabs/motors.html]: class should be "togglesmall" instead of "toggle"

**Comment by skaman82** [comment]
@NightHawk32 could you please review and test this too?

**Comment by tricopterY** [comment]
I just noticed if feature 3D mode is configured, the 3D neutral should automatically be 1500 or 1520 for Futaba radios. Otherwise the motors will default/jump to max 1000 in reverse, once the test mode is checked.
Let me look at this again.

**Comment by skaman82** [comment]
In my test the 3D neutral equals the 3D low setting in config tab.

**Comment by skaman82** [comment]
With older firmware the motor test switch should be blocked if 3D is enabled. That also works as it should with 1.10 with your version.

**Comment by tricopterY** [comment]
```javascript
<div class="number">
                        <label> <input type="number" name="3ddeadbandlow" step="1" min="1000" max="2000" /> <span
                            i18n="configuration3dDeadbandLow"></span>
                        </label>
                    </div>
                    <div class="number">
                        <label> <input type="number" name="3ddeadbandhigh" step="1" min="1000" max="2000" /> <span
                            i18n="configuration3dDeadbandHigh"></span>
                        </label>
                    </div>
```
I reckon these are way off - should be something like:
1500 - 1575 for High and 1425 - 1500 for Low, and relative to 3D-Neutral
Otherwise:
```javascript
if( (MOTOR_DATA[i] < _3D.deadband3d_low) || (MOTOR_DATA[i] > _3D.deadband3d_high) ){
                    motors_running = true;
                    break;
                }
```
will enable:
```javascript
$('#motorsEnableTestMode').prop('checked', true);
```
every time the tabs are switched.

**Comment by NightHawk32** [comment]
@tricopterY nice work, looks good.
Did some testing, now this works as expected.
The values for deadband could extented to min 1400 for low and max 1600 for high. This shouldn't be needed for flying but can be useful to do some experimenting about the reversing speed of the motors.

**Comment by tricopterY** [comment]
To check the logic and firing of motors, motors.js line#356:
```javascript
MSP.send_message(MSP_codes.MSP_SET_MOTOR, buffer);
GUI.log("motors firing "+MOTOR_DATA[0]+"- "+MOTOR_DATA[1]+"- "+MOTOR_DATA[2]+"- "+MOTOR_DATA[3]);
```
Each time tabs are switched the motors fire with the value of _3D.neutral3d. The danger is when the deadbands enable the test mode and sliders. The fix is to limit the 3D deadbands.

**Comment by NightHawk32** [comment]
I didn't  notice that since I have applied a fix for that in my dev version.
Anyway I don't think just limiting the deadbands is enough. In my opinion you shouldn't write any value to the motors unless you can make sure that you know how to turn them off. So if msp 3D isn't supported the motor values shouldn't be modified, since they were set via CLI and therefore your limits won't affect them. I did some experiments about this over here: #319

**Comment by tricopterY** [comment]
I stumbled on this when I enabled 3D and had the 3d-Neutral at 1520 (futaba radios neutral) and 3D-Deadbands at their CLI defaults.
The testMode checkbox got enabled as soon as I switched to Motors tab.
```javascript
if( (MOTOR_DATA[i] < _3D.deadband3d_low) || (MOTOR_DATA[i] > _3D.deadband3d_high) ){
                    motors_running = true;
                    break;
                }
```
Thanks for the fix.
Not sure if the CLI defaults should also be revisited.

**Comment by NightHawk32** [comment]
Yes we should think about some proper values for 3D defaults.
I merged this.

**Pull request merged**